### ISSUE TITLE
Some questions about MVDR  Implementation in espnet

### ISSUE THREAD

**Issue by Rpersie** [espnet/espnet#992]
Recently, I am working to improve the espnet beamformer. I want to implement the MVDR beamformer. In MVDR beamformer, we will encounter the  Eigenvalue decomposition.  Although the pytroch can not support complex Eigenvalue decomposition, the psd_s is  an  Hermite matrix. So we can use the real part and image part to construct a new real Symmetric matrix. Besides , the pytorch support the real Symmetric matrix decomposition. 
I have completed the training. And the train-loss, train-acc, vaild-loss and vaild-acc is better than espnet. But the test results is too bad. Do you think this idea is right? How should I debug this.
Thank you! @kamo-naoyuki @sw005320

[traineigen.log](https://github.com/espnet/espnet/files/3440554/traineigen.log)
[trainespnet.log](https://github.com/espnet/espnet/files/3440555/trainespnet.log)
![result](https://user-images.githubusercontent.com/12673020/62026901-e7d08700-b20e-11e9-94c7-19c4a4891ecd.png)

**Comment by Rpersie** [comment]
I verify the eigenvalue decomposition in matlab and it is ok !  Besides I use the pytorch torch.symeig to do the  eigenvalue decomposition

**Comment by kamo-naoyuki** [comment]
I often met such case that val-acc is better but the CER is worse with this task. I think one of the reason is the mismatch variability between dev set and eval set.
Your result looks better, or near, than the current implementation about dt, so that is not necessary due to bugs.

As general advise, it's better to test the gradient of eigen decomposition, but I think it may be O.K. I have also implemented the same implementation once with same way as you said.

**Comment by Rpersie** [comment]
Thank you ! You results is better than the original beamformer?

**Comment by kamo-naoyuki** [comment]
Sorry, I have not tested it on ESPNet.

**Comment by Rpersie** [comment]
In my Implementation, I also encounter an problem. That is described in https://github.com/pytorch/pytorch/issues/23133
I just use this trick to do Eigenvalue decomposition, otherwise the torch.symeig() is not stable. 
Do you think this trick is right ?  Thank you!
